File: location/src/main/kotlin/com/sleekydz86/location/application/location/Coordinates.kt
package com.sleekydz86.location.application.location

data class Coordinates(
    val latitude: Double,
    val longitude: Double
) {
    init {
        require(latitude in -90.0..90.0) { "위도는 -90에서 90 사이여야 합니다." }
        require(longitude in -180.0..180.0) { "경도는 -180에서 180 사이여야 합니다." }
    }
}

File: location-mobile/src/domain/value-objects/Coordinates.ts
export interface Coordinates {
  readonly latitude: number;
  readonly longitude: number;
}

export function createCoordinates(latitude: number, longitude: number): Coordinates {
  if (latitude < -90 || latitude > 90 || longitude < -180 || longitude > 180) {
    throw new RangeError('잘못된 좌표입니다.');
  }
  return { latitude, longitude };
}
